sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java

```diff
@@ -32,12 +32,21 @@
   private final String[] qualifiers;
   private final String language;
 
-  public ComponentQuery(@Nullable String nameOrKeyQuery, String language, String... qualifiers) {
-    this.language = language;
-    checkArgument(qualifiers.length > 0, "At least one qualifier must be provided");
-
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Dev Cockpit doesn't use it anymore
+   */
+  @Deprecated
+  public ComponentQuery(@Nullable String nameOrKeyQuery, String... qualifiers) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)